### PR TITLE
Fix: Fix Ray WFs failing caused by any task returning dict defined in return statement

### PR DESCRIPTION
**Pull request by SebastianMorawiec**
# The problem

When the task had dict defined in return statement it caused the whole workflow to fail. Caused by the fact that we treated dicts as subscribtable (with [] operator) which they are not.

# This PR's solution

Do not treat dict return values as subscribtable

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/imp[rovement]/int[ernal]/docs>[!]:`
- [x] The PR is linked to a JIRA ticket (if there's no suitable ticket, check the box).

### CODE REVIEW

File: src/orquestra/sdk/_base/_ast.py [hunk 1]
```diff
@@ -114,9 +114,7 @@ def visit_Dict(self, node):
             # having a single `None` output.
             self.outputs.add(_AstReturnMetadata(is_subscriptable=False, n_outputs=1))
         else:
-            self.outputs.add(
-                _AstReturnMetadata(is_subscriptable=True, n_outputs=n_elements)
-            )
+            self.outputs.add(_AstReturnMetadata(is_subscriptable=False, n_outputs=1))
```
Comment (jamesclark-Zapata) [inline src/orquestra/sdk/_base/_ast.py:117]: **suggestion:** ‏Can you add a comment explaining why we're doing this?
Both sides of the if statement are the same, which may look odd without any explanation.

File: tests/runtime/ray/test_integration.py [hunk 2]
```diff
@@ -926,3 +926,32 @@ def test_outside_workflow(self):
...
+@pytest.mark.slow
+class TestDictReturnValue:
+    """
+    Tasks that had dicts in return statement used to cause WF failures.
+    This test might look trivial bul unless AST code was majorly refactored,
+    please abstain from removing it (also refer to git history for the fix itself)
+    """
+
+    def test_dict_as_task_return_value(self, runtime: _dag.RayRuntime):
+        @sdk.task(source_import=sdk.InlineImport())
```
Comment (jamesclark-Zapata) [inline tests/runtime/ray/test_integration.py:940]: **suggestion:** ‏
This is the default now.
```suggestion
        @sdk.task
```

**Review by jamesclark-Zapata — APPROVED** [review]
LGTM with some minor comments

File: src/orquestra/sdk/_base/_ast.py (hunk 1 above)
Comment (SebastianMorawiec) [reply src/orquestra/sdk/_base/_ast.py:117]: Imma just colapse them into one statement for now

**Pull request merged**